src/memray/commands/run.py

```diff
@@ -237,18 +252,24 @@ def validate_target_file(self, args: argparse.Namespace) -> None:
         if args.run_as_module:
             return
         try:
-            source = pathlib.Path(args.script).read_bytes()
+            if args.run_as_cmd:
+                source = bytes(args.script, "UTF-8")
+            else:
+                source = pathlib.Path(args.script).read_bytes()
             ast.parse(source)
         except (SyntaxError, ValueError):
             raise MemrayCommandError(
-                "Only Python files can be executed under memray", exit_code=1
+                "Only valid Python files or commands can be executed under memray",
+                exit_code=1,
             )
 
     def run(self, args: argparse.Namespace, parser: argparse.ArgumentParser) -> None:
         if args.live_port is not None and not args.live_remote_mode:
             parser.error("The --live-port argument requires --live-remote")
         if args.follow_fork is True and (args.live_mode or args.live_remote_mode):
             parser.error("--follow-fork cannot be used with the live TUI")
+        if args.run_as_cmd and args.script.endswith(".py"):
```

Comment:
I like the user friendliness of this, but it's not technically correct - a command _can_ end with `.py` and still be valid - for instance:
```shell
python -c 'from types import SimpleNamespace as NS; ns = NS(py=1); ns.py'
```
is a perfectly valid command to run. Instead, how about if we use:
```suggestion
        if args.run_as_cmd and pathlib.Path(args.script).exists():
```
to check if the script they've given is the name of a file that exists on disk? Technically that file name could _also_ be a valid Python command to execute, but it's even less likely, and it would also handle cases where the filename doesn't end in `.py`

Reply:
yeah, could happen with a shebang. i use it less nowadays. anyway, exists() doesn't throw exceptions, so seems ok.